### ISSUE TITLE
call path & removing 

### ISSUE THREAD

**Issue by IwanKaramazow** [ekosz/redux-falcor#6]
I noticed an issue when removing items from a list when calling a path like, e.g.  ````userList.list.remove //removing user by id````
The reducers gets the following response:
````javascript
 {
   "json": {
      "userList": {
         "length": 2
      }
   }
}
````
while my datasource returns:
````javascript
[{
        path: ['userList', 'list', args[1]],
        invalidated: true   
       },
       {
        path: ['userList', 'length'],
        value: newLength
 }]
````

The problem is probably the merge that happens when the reducer receives a success response.
The reducer has no idea something has been invalidated in the cache, because the response only shows  the new length.

Only solution I can think of right now is, when calling a path, remove all entities and put the cache from the model under entities. This is probably horribly inefficient, if you have to delete and retrieve the ***whole*** cache...

Any ideas?

**Comment by ekosz** [comment]
Yeah I ran into this the other day as well. redux-falcor works best when items are only being added / changed to the cache, but has no way of detecting caching invalidations currently.

I have [an open issue](https://github.com/Netflix/falcor/issues/572) (albit old now) with the falcor team about how to keep the redux cache more inline with falcor. 

As a workaround the best way to remove the item from the redux cache would be to explicitly return a null atom where the old item used to be.

So your router would need to returning something like:

```js
[{
        path: ['userList', 'list', 2],
        value: { $type: atom },   
       },
       {
        path: ['userList', 'length'],
        value: newLength
 }]
```

That *should* remove the item from redux.

**Comment by IwanKaramazow** [comment]
The above isn't currently working, ran into some bugs with falcor (again ;) ).

In theory the cache on the client should have the valid model:  for now I'll try to clear the slice of state I wanna update through a falcor call & do a ```dispatch(retrievePath())``` to get the valid model back in....

This is probably a horrible idea.

**Comment by IwanKaramazow** [comment]
I'll take a look at this, when I somehow get Falcor's ```call``` working again. At the moment every call silently destroys my model's cache 😅

Falcor is going to be a lot more pain than I thought, even when I get around all the bugs...

**Comment by ekosz** [comment]
This should be fixed in version 3.0 when it's released.
